src/Estimators/EstimatorManagerNew.cpp

```diff
@@ -101,6 +102,7 @@ EstimatorManagerNew::EstimatorManagerNew(Communicate* c,
     if (!(createEstimator<SpinDensityInput>(est_input, pset.getLattice(), pset.getSpeciesSet()) ||
           createEstimator<MomentumDistributionInput>(est_input, pset.getTotalNum(), pset.getTwist(),
                                                      pset.getLattice()) ||
+          createEstimator<SelfHealingOverlapInput>(est_input,twf) ||
```

Comment:
@PDoakORNL This feature seems requiring MSD, how to implement a trap? I would like to avoid implementing the trap in the constructor but right before calling the constructor.

Reply:
The estimator actually doesn't require MSD per se, as actually any wavefunction that involves a linear combination will do.  In the interest of future-proofing, I would prefer to limit how much hard coding is done to forever bind this estimator to a single, now extant class.

Reply:
If there is a second WF with linear combination, we should introduce another accessor base class for all the WF with the same feature and dynamic_cast to this base class. Again, we still need checks before constructing the estimator object. I'd like to know how to inject such checks, current `createEstimator` doesn't facilitate that. @PDoakORNL any input, I'm thinking of not using `createEstimator` and just expand `createEstimator` calls explicitly.

Reply:
Ideally you should be resolving that sort of defective in the input classes. If you can't figure out such a thing out without tearing apart the simulation time twf in estimator specific way then if makes sense for the estimators constructor to throw an exception which is going to lead to a fatal error. I don't think estimator specific error checks should be anywhere except in a particular estimator class. I don't see any issue with a constructor throwing for such an issue.

Reply:
I usually prefer not to throw from a constructor because it may leave half baked stuff and cannot call destructor. probably it is OK here since anyway it will be a fatal exception that leads to abort.

Reply:
I don't support throwing nonfatal exceptions from a constructor either but in general our convention is to not throw unless its fatal. This keeps the code in the one of the two acceptable places to have it.